src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBExtensions.cs

```diff
@@ -298,15 +298,25 @@ public static IResourceBuilder<AzureCosmosDBEmulatorResource> WithDataExplorer(t
             throw new NotSupportedException($"The Data Explorer endpoint is only available when using the preview version of the Azure Cosmos DB emulator. Call '{nameof(RunAsPreviewEmulator)}' instead.");
         }
 
-        return builder.WithEndpoint(endpointName: "data-explorer", endpoint =>
-        {
-            endpoint.UriScheme = "http";
-            endpoint.TargetPort = 1234;
-            endpoint.Port = port;
-        });
+        var endpointName = "data-explorer";
+        return
+            builder.WithEndpoint(endpointName: endpointName, endpoint =>
+            {
+                endpoint.UriScheme = "http";
+                endpoint.TargetPort = 1234;
+                endpoint.Port = port;
+            })
+            .WithUrls(context =>
+            {
+                var url = context.Urls.FirstOrDefault(u => u.Endpoint?.EndpointName == endpointName);
+                if (url is not null)
+                {
+                    url.DisplayText = "Data Explorer";
```

Comment:
We would need to justify not having this value localized or provide a relatively simple way for users to set localized values

Reply:
For known endpoints like this, we should override the display text on the dashboard with a localized value like is done in `src/Aspire.Dashboard/ResourceService/Partials.cs#124` with known commands.